src/app/loadout-drawer/loadout-drawer-reducer.ts

```diff
@@ -56,6 +58,36 @@ import {
  */
 export type LoadoutUpdateFunction = (loadout: Loadout) => Loadout;
 
+/** Some helpers that bind our updater functions to the current environment */
+export function useLoadoutUpdaters(
+  store: DimStore,
+  setLoadout: (updater: LoadoutUpdateFunction) => void
+) {
+  const defs = useD2Definitions()!;
+
+  function useUpdater<T extends unknown[]>(fn: (...args: T) => LoadoutUpdateFunction) {
+    return useCallback((...args: T) => setLoadout(fn(...args)), [fn]);
+  }
+  function useDefsUpdater<T extends unknown[]>(
+    fn: (defs: D1ManifestDefinitions | D2ManifestDefinitions, ...args: T) => LoadoutUpdateFunction
+  ) {
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
This seems like an instance of https://github.com/facebook/react/issues/25958 -- can we at least add a link and/or a short explanation as to why react-hooks considers these dependencies erroneous?

Reply:
Yeah, it looks like the pattern of "a hook that returns hooks" is not accounted for by the lint. I added some comments.